Source/WebCore/Modules/applepay/PaymentInstallmentConfiguration.mm

```diff
@@ -194,72 +208,80 @@ static PKInstallmentRetailChannel platformRetailChannel(ApplePayInstallmentRetai
         return configuration;
 
     [configuration setInstallmentItems:createNSArray(coreConfiguration.items).get()];
-    [configuration setApplicationMetadata:applicationMetadata];
+    [configuration setApplicationMetadata:applicationMetadataDictionary(coreConfiguration).get()];
     [configuration setRetailChannel:platformRetailChannel(coreConfiguration.retailChannel)];
 
     return configuration;
 }
 
 ExceptionOr<PaymentInstallmentConfiguration> PaymentInstallmentConfiguration::create(const ApplePayInstallmentConfiguration& configuration)
 {
-    NSDictionary *applicationMetadataDictionary = nil;
-    if (!configuration.applicationMetadata.isNull()) {
-        NSData *applicationMetadata = [configuration.applicationMetadata dataUsingEncoding:NSUTF8StringEncoding];
-        applicationMetadataDictionary = dynamic_objc_cast<NSDictionary>([NSJSONSerialization JSONObjectWithData:applicationMetadata options:0 error:nil]);
-        if (!applicationMetadataDictionary)
-            return Exception { TypeError, "applicationMetadata must be a JSON object"_s };
-    }
+    auto dictionary = applicationMetadataDictionary(configuration);
+    if (!configuration.applicationMetadata.isNull() && !dictionary)
+        return Exception { TypeError, "applicationMetadata must be a JSON object"_s };
 
-    return PaymentInstallmentConfiguration(configuration, applicationMetadataDictionary);
+    return PaymentInstallmentConfiguration(ApplePayInstallmentConfiguration(configuration), WTFMove(dictionary));
 }
 
-PaymentInstallmentConfiguration::PaymentInstallmentConfiguration(const ApplePayInstallmentConfiguration& configuration, NSDictionary *applicationMetadata)
-    : m_configuration { createPlatformConfiguration(configuration, applicationMetadata) }
+static ApplePayInstallmentConfiguration addApplicationMetadata(const ApplePayInstallmentConfiguration& input, RetainPtr<NSDictionary>&& applicationMetadata)
 {
+    auto configuration = input;
```

Comment:
Nit: could probably simplify this by passing the `ApplePayInstallmentConfiguration` by value, but this is OK as-is.